pkg/scalers/scalersconfig/typed_config.go

```
@@ -0,0 +1,402 @@
+	split := strings.Split(valFromConfig, elemSeparator)
+	for i, s := range split {
+		s := strings.TrimSpace(s)
+		if err := setConfigValueHelper(s, reflect.ValueOf(elemIfc).Elem()); err != nil {
+			return fmt.Errorf("slice element %d: %w", i, err)
+		}
+		field.Set(reflect.Append(field, reflect.ValueOf(elemIfc).Elem()))
+	}
+	return nil
+}
+
+// setParamValueHelper is a function that sets the value of the parameter
+func setConfigValueHelper(valFromConfig string, field reflect.Value) error {
+	paramValue := reflect.ValueOf(valFromConfig)
+	if paramValue.Type().AssignableTo(field.Type()) {
+		field.SetString(valFromConfig)
+		return nil
+	}
+	if paramValue.Type().ConvertibleTo(field.Type()) {
+		field.Set(paramValue.Convert(field.Type()))
+		return nil
+	}
+	if field.Type() == reflect.TypeOf(url.Values{}) {
+		return setConfigValueURLParams(valFromConfig, field)
+	}
+	if field.Kind() == reflect.Map {
+		return setConfigValueMap(valFromConfig, field)
+	}
+	if field.Kind() == reflect.Slice {
+		return setConfigValueSlice(valFromConfig, field)
+	}
+	if field.CanInterface() {
+		ifc := reflect.New(field.Type()).Interface()
+		if err := json.Unmarshal([]byte(valFromConfig), &ifc); err != nil {
+			return fmt.Errorf("unable to unmarshal to field type %v: %w", field.Type(), err)
+		}
+		field.Set(reflect.ValueOf(ifc).Elem())
+		return nil
+	}
+	return fmt.Errorf("unable to find matching parser for field type %v", field.Type())
+}
+
+// configParamValue is a function that returns the value of the parameter based on the parsing order
+func (sc *ScalerConfig) configParamValue(params Params) (string, bool) {
+	for _, po := range params.ParsingOrder {
+		var m map[string]string
+		key := params.Name
+		switch po {
+		case TriggerMetadata:
+			m = sc.TriggerMetadata
+		case AuthParams:
+			m = sc.AuthParams
+		case ResolvedEnv:
+			m = sc.ResolvedEnv
+			key = sc.TriggerMetadata[fmt.Sprintf("%sFromEnv", params.Name)]
+		default:
+			m = sc.TriggerMetadata
```

I'm not sure whether its appropriate to set this in the default case :thinking: Could you explain more about this ?

---

that is a good point, now looking at it, this should probably throw an error instead because `ParsionOrder` should only be one of `TriggerMetadata`, `AuthParams` or `ResolvedEnv`.